### PR TITLE
Add keypair validation

### PR DESCRIPTION
**Pull request by szopen321**
proposed fix for https://github.com/spulec/moto/issues/5151

### CODE REVIEW

**Comment by szopen321** [comment]
This PR is a WIP. 
I'm creating it to show my aproach and possibly change it according to your suggestions.

File: tests/test_ec2/test_ec2_integration.py [hunk 1]
```diff
@@ -11,13 +11,16 @@ def test_run_instance_with_encrypted_ebs():
     kms = boto3.client("kms", region_name="us-east-1")
     resp = kms.create_key(Description="my key", KeyUsage="ENCRYPT_DECRYPT")
     key_id = resp["KeyMetadata"]["Arn"]
+    ec2_client = boto3.client("ec2", region_name="us-east-1")
```
Comment (bblommers) [inline tests/test_ec2/test_ec2_integration.py:14]: This client is already initialized two lines down

File: moto/ec2/models/instances.py [hunk 2]
```diff
@@ -550,6 +550,8 @@ def get_instance(self, instance_id):
     def add_instances(self, image_id, count, user_data, security_group_names, **kwargs):
         location_type = "availability-zone" if kwargs.get("placement") else "region"
         default_region = "us-east-1"
+        describe_key_pairs = self.describe_key_pairs
+        describe_key_pairs(key_names=[kwargs.get("key_name")])
```
Comment (bblommers) [inline moto/ec2/models/instances.py:554]:
```suggestion
        self.describe_key_pairs(key_names=[kwargs.get("key_name")])
```
Minor nitpick - I don't think reassigning the method does anything for readability

**Review by bblommers — CHANGES_REQUESTED** [review]
Thanks for the PR @szopen321! The approach looks good, just left a few comments

**Comment by szopen321** [comment]
I have put keypair validation behind the flag as it can cause problems potentially.

**Comment by bblommers** [comment]
> I have put keypair validation behind the flag as it can cause problems potentially.

That actually makes a lot of sense @szopen321.

Now that this validation is optional, we wouldn't need to change everything - existing tests can continue to use an invalid keypair. Would you mind changing the PR to only include he logic and the one new test, that validates that it breaks with the validation enabled?
I realize that it's a bit of extra work - feel free to just open a new branch/PR with only the required changes if that's easier.

**Review by bblommers — APPROVED** [review]
Awesome - thanks @szopen321!

**Pull request merged**